### PR TITLE
Apply shave bit to tile space for histroy output

### PR DESCRIPTION
**Pull request by weiyuan-jiang**
When a history collection in tile space is binary, the "nbits" is not used. The PR adds that function. It will be zero-diff for test but will be non-zero diff for binary history output.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (affects only documentation or cleanup)

## Checklist:

- [ ] I have tested this change with a run of GEOSgcm (if non-trivial)
- [x] I have added one of the required labels (0 diff, 0 diff trivial, 0 diff structural, non 0-diff)
- [x ] I have updated the CHANGELOG.md accordingly following the style of [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

### CODE REVIEW

**Comment by bena-nasa** [comment]
Could you make this a subroutine call in the gridcomp rather than just a new block of code in the run routine? Maybe just a subroutine that takes a state and does the bit shaving, that seems cleaner.

**Comment by weiyuan-jiang** [comment]
good suggestion. The block has been moved to a subroutine @bena-nasa

**Review by bena-nasa — APPROVED** [review]

File: gridcomps/History/MAPL_HistoryGridComp.F90 [hunk 1]
```diff
@@ -38,7 +38,7 @@ module MAPL_HistoryGridCompMod
   use mapl_RegridMethods
   use MAPL_newCFIOitemVectorMod
   use MAPL_newCFIOitemMod
-  use pFIO_ClientManagerMod, only: o_Clients
+  use pFIO
```
Comment (mathomp4) [inline gridcomps/History/MAPL_HistoryGridComp.F90:41]: Just a question: do we need to use all of pFIO here? Or just one routine a la:
```fortran
use pFIO, only: shavebits
````

I'm not sure what @tclune thinks of this. I don't see anything in the [Style Guide](https://github.com/GEOS-ESM/MAPL/wiki/MAPL-Coding-standards) for how to handle use/use-only stuff...

Comment (mathomp4) [reply gridcomps/History/MAPL_HistoryGridComp.F90:41]: Oh I'm an idiot. it's in the file. Time to end the day! :)

Comment (weiyuan-jiang) [reply gridcomps/History/MAPL_HistoryGridComp.F90:41]: I am making changes to use smallest set of routines from pfio

**Review by mathomp4 — APPROVED** [review]
Approved

**Pull request merged**